python/ray/_private/runtime_env/packaging.py

```diff
@@ -217,21 +169,19 @@ def parse_uri(pkg_uri: str) -> Tuple[Protocol, str]:
             f"Invalid protocol for runtime_env URI {pkg_uri}. "
             f"Supported protocols: {Protocol._member_names_}. Original error: {e}"
         )
-    if protocol == Protocol.S3 or protocol == Protocol.GS:
-        return (protocol, f"{protocol.value}_{uri.netloc}{uri.path.replace('/', '_')}")
-    elif protocol == Protocol.HTTPS:
-        parsed_netloc = uri.netloc.replace(".", "_").replace(":", "_").replace("@", "_")
-        return (
-            protocol,
-            f"https_{parsed_netloc}{uri.path.replace('/', '_')}",
-        )
-    elif protocol == Protocol.FILE:
-        return (
-            protocol,
-            f"file_{uri.path.replace('/', '_')}",
-        )
-    else:
-        return (protocol, uri.netloc)
+
+    package_name = uri.netloc
+    if protocol in Protocol.remote_protocols():
+        package_name = f"{protocol.value}_{uri.netloc}{uri.path}"
+
+    disallowed_chars = ["/", ":", "@", "+"]
+    for disallowed_char in disallowed_chars:
+        package_name = package_name.replace(disallowed_char, "_")
+
+    # Remove all periods except the last one, which is used for the file extension
```

Comment:
Do we need to worry about extensions with multiple dots like `.tar.gz` or does it not matter somehow because we only deal with `.zip`?

Reply:
That's a great point– currently [we validate](https://github.com/ray-project/ray/blob/275dd70a1bc1f71867666e20fb28168ac8104832/python/ray/_private/runtime_env/validation.py#L29) that the files are `.zip` files, but we should consider expanding that to allow other types of zip files.

I think for now it's okay to assume we only deal with `.zip` because of the pre-existing validation.